### PR TITLE
Fix #9044 - Add prometheus event monitor

### LINKED ISSUE #9044: Send pipeline status metric to Prometheus
create a new eventMonitor for prometheus

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #9044

Added the Prometheus Event monitor using the current Micrometer Bundle that we are registering for Data Insights metrics.

We are creating a new counter `ingestionPipeline.counter` with different labels by cluster name, FQN, and event types for the Ingestion Pipelines lifecycle.

It looks like follows:

```
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="running",fqn="mysql.mysql_metadata_1R2OCGwI",pipelineType="metadata",} 2.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="success",fqn="mysql.mysql_metadata_1R2OCGwI",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="running",fqn="mysql.mysql_metadata_uzP1T419",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="entityDeleted",fqn="mysql.mysql_metadata_MAhqvRxU",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="entityDeleted",fqn="mysql.mysql_metadata_RYAdtGIT",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="entityCreated",fqn="mysql.mysql_metadata_RYAdtGIT",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="entityDeleted",fqn="mysql.mysql_metadata_X25TBRNp",pipelineType="metadata",} 1.0
ingestionPipeline_counter_total{clusterName="openmetadata-pere-local",eventType="running",fqn="bad_mysql.bad_mysql_metadata_N8dPqQqu",pipelineType="metadata",} 2.0
```

Thanks

#
### Type of change :

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: conf/openmetadata.yaml [hunk 1]
```diff
@@ -171,7 +171,7 @@ elasticsearch:
   batchSize: ${ELASTICSEARCH_BATCH_SIZE:-10}
 
 # eventMonitoringConfiguration:
-  # eventMonitor: ${EVENT_MONITOR:-cloudwatch}  # Possible values are "cloudwatch"
+  # eventMonitor: ${EVENT_MONITOR:-prometheus}  # Possible values are "prometheus", "cloudwatch"
```
Comment (pmbrull) [inline conf/openmetadata.yaml:174]: setting prometheus as default, if enabled

File: openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   * A new counter will be created only if it does not exist for the
+   * given set of tags. Otherwise, micrometer will increase the count
+   * of the existing counter.
+   * Ref <a href="https://stackoverflow.com/questions/59592118/dynamic-tag-values-for-the-counter-metric-in-micrometer">...</a>
+   *
+   * @param fqn Pipeline FQN
+   * @param pipelineType Metadata, Profiler,...
+   * @param eventType running, failed, entityCreated,...
+   */
+  public void incrementIngestionPipelineCounter(String fqn, String pipelineType, String eventType) {
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java:85]: dynamically building the counter based on the Change Event

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java:84]: @pmbrull looks great. Do we have any other data capturing here, like no.of entities that are successfully ingested vs failed etc..

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java:84]: also any strings such as "pipelineStatus" lets make private final static

Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/monitoring/PrometheusEventMonitor.java:84]: We're not yet sending the `summary` of the workflow as a pipeline status. Once we apply that change, we can update the Change Events to track those statistics as well. Let me open a separate ticket for that

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**